### PR TITLE
[#50] Trigger a compil. failure on missing @​Self

### PR DESCRIPTION
**Pull request by fbiville**
This used to crash without any error message.

### CODE REVIEW

File: jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java [hunk 1]
```diff
@@ -87,12 +107,24 @@ public boolean process(Set<? extends TypeElement> annotations,
...
-        tryExportGraph(roundEnv);
+        if (validateMappings(roundElements)) {
+            tryGenerateSources(roundElements);
+            tryExportGraph(roundEnv);
+        }
 
         return false;
     }
 
+    private boolean validateMappings(Multimap<ClassName, Mapping> roundElements) {
```
Comment (jcgay) [inline jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java]: Could be extracted in its own class with `classesWithoutSelf` ?

File: jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java [hunk 2]
```diff
@@ -87,12 +107,24 @@ public boolean process(Set<? extends TypeElement> annotations,
...
+        Collection<ClassName> classesWithoutSelf = classesWithoutSelf(roundElements);
+        if (!classesWithoutSelf.isEmpty()) {
```
Comment (jcgay) [inline jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java]: Purely cosmetic but I would have inverted the `if`.

File: jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java [hunk 3]
```diff
@@ -124,15 +156,15 @@ private void tryGenerateContextPathHolder() throws IOException {
 
     private void generateLinkerSources(Multimap<ClassName, Mapping> elements) throws IOException {
         for (ClassName className : elements.keySet()) {
-            generateLinkerClasses(className, elements.get(className), CONTEXT_PATH_HOLDER);
+            generateLinkerClasses(className, elements.get(className));
```
Comment (jcgay) [inline jax-rs-linker/src/main/java/fr/vidal/oss/jax_rs_linker/LinkerAnnotationProcessor.java]: Nicely spot :)

**Comment by jcgay** [comment]
:+1:

**Comment by fbiville** [comment]
All fixed in cc818a419b73aeb970f3ae0930d1ecfb2b21b114.

**Comment by jcgay** [comment]
Merged in 26ced8f.

**Pull request closed without merging**